### PR TITLE
T-51 Changes to asset controller

### PR DESCRIPTION
**Pull request by DashaDrahun**
refactored asset contoller's methods, moved them to service, added and refactored tests

### CODE REVIEW

File: PortfolioKeeper/Controllers/AssetController.cs [hunk 1]
```diff
@@ -85,82 +70,50 @@ public ActionResult GetAssetPropertiesTypes(byte parentTypeId, byte childrenType
...
-
-            var dtoCollection = assetService.GetAssetProperties(id);
-            var viewModelProperties = mapper.Map<IEnumerable<AssetPropertyValueDto>, IEnumerable<AssetPropertyValueViewModel>>(dtoCollection);
-            var model = new AssetDetailsViewModel
-            {
-                Id = id,
-                Properties = viewModelProperties.ToList()
-            };
+            securityService.CheckAssetAccess(id, User.Identity.GetUserId());
+            var model = assetService.GetDetails(id);
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AssetController.cs]: empty line before var

File: PortfolioKeeper.Tests/BusinessLogic/Services/SecurityCheckServiceTests.cs [hunk 2]
```diff
@@ -119,5 +122,15 @@ public void CheckAssetAccess_When_No_Permissions_Should_Throw_SecurityException(
             // assert
             act.Should().Throw<SecurityException>();
         }
+
+        [Test]
+        public void CheckAssetAccess_UserId_Is_Null_Should_Throw_SecurityException()
```
Comment (goremukin) [inline PortfolioKeeper.Tests/BusinessLogic/Services/SecurityCheckServiceTests.cs]: When_UserId

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 3]
```diff
@@ -46,18 +49,20 @@ public class AssetService : IAssetService
...
             this.assetTypeMapper = assetTypeMapper
                                ?? throw new ArgumentNullException(nameof(assetTypeMapper));
-            this.propertyTypeValueMapper = propertyTypeValueMapper
-                                   ?? throw new ArgumentNullException(nameof(propertyTypeValueMapper));
+            this.propertyTypeValueMapper = propertyMapper
+                                   ?? throw new ArgumentNullException(nameof(propertyMapper));
             this.assetMapper = assetMapper
                           ?? throw new ArgumentNullException(nameof(assetMapper));
+            this.assetHelper = assetHelper
+                          ?? throw new ArgumentNullException(nameof(assetHelper));
```
Comment (goremukin) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: remove one empty line

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 4]
```diff
@@ -212,6 +217,26 @@ public void Create(List<AssetPropertyValueViewModel> properties, string userId,
             unitOfWork.Commit();
         }
 
+        public void AddConcreteAsset<T>(T asset, byte childAssetTypeId, string userId) where T : BaseAssetViewModel
+        {
+            var propertyTypes = GetPropertyTypes(childAssetTypeId);
+            var allPropertyValues = assetHelper.GetPropertyValuesFromTypes(asset, propertyTypes);
+            Create(allPropertyValues, userId, asset.Name, childAssetTypeId);
```
Comment (goremukin) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: add empty line before Create, please

**Pull request merged**